libvast/src/system/exporter.cpp

```diff
@@ -199,11 +200,21 @@ void handle_batch(exporter_actor::stateful_pointer<exporter_state> self,
 } // namespace
 
 exporter_actor::behavior_type
-exporter(exporter_actor::stateful_pointer<exporter_state> self, expression expr,
-         query_options options, std::vector<pipeline>&& pipelines) {
+exporter(exporter_actor::stateful_pointer<exporter_state> self,
+         std::optional<expression> expr,
+         std::vector<std::string> spawn_arguments, query_options options,
+         std::vector<pipeline>&& pipelines) {
+  // Parse given expression.
+  if (!expr) {
+    auto normalized_expr = system::get_expression(spawn_arguments);
+    if (!normalized_expr) {
+      self->quit(normalized_expr.error());
+    }
+    expr.emplace(*normalized_expr);
+  }
```

Comment:
I believe this should only normalize and validate (see the above review comment).

After you quit you need to `return exporter_actor::behavior_type::make_empty_behavior()`, otherwise you immediately afterwards dereference an invalid expression.

I would also recommend enriching the error with some additional context, e.g.,

```c++
self->quit(caf::make_error(
  ec::format_error,
  fmt::format("exporter failed to normalize and validate expression: {}",
    normalized_expr.error())));
```